### PR TITLE
Add 7splus keyboard.

### PR DESCRIPTION
**Pull request by Salicylic-acid3**
A 85 keys ANSI Layout split keyboard.
Salicylic-acid3

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/7splus/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          }
+        break;
+    #endif
+    default:
+      result = true;
+      break;
+  }
+
+  return result;
+}
```
Comment (drashna) [inline keyboards/7splus/keymaps/default/keymap.c:92]: There is a lot of unused code here, or is unnecessary. 

```suggestion
bool process_record_user(uint16_t keycode, keyrecord_t *record) {
    switch (keycode) {
#ifdef RGBLIGHT_ENABLE
        case RGB_RST:
            if (record->event.pressed) {
                uint8_t mode = rgblight_get_mode();
                eeconfig_update_rgblight_default();
                rgblight_enable();
                rgblight_mode(mode);
             }
             break;
#endif
    }

    return true;
}
```

File: keyboards/7splus/keymaps/salicylic/keymap.c [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+          }
+        break;
+    #endif
+    default:
+      result = true;
+      break;
+  }
+
+  return result;
+}
```
Comment (drashna) [inline keyboards/7splus/keymaps/salicylic/keymap.c:143]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
    return update_tri_layer_state(state, _RAISE, _LOWER, _ADJUST);
}

bool process_record_user(uint16_t keycode, keyrecord_t *record) {
    switch (keycode) {
#ifdef RGBLIGHT_ENABLE
        case RGB_RST:
            if (record->event.pressed) {
                uint8_t mode = rgblight_get_mode();
                eeconfig_update_rgblight_default();
                rgblight_enable();
                rgblight_mode(mode);
             }
             break;
#endif
    }
    return true;
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/7splus/rules.mk [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (fauxpark) [inline keyboards/7splus/rules.mk:22]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/7splus/keymaps/salicylic/keymap.c [hunk 4]
```diff
@@ -0,0 +1,130 @@
+#include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline keyboards/7splus/keymaps/salicylic/keymap.c:1]: This file should have a license header, as the default keymap does.

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/7splus/keymaps/salicylic/keymap.c (hunk 4 above)
Comment (Salicylic-acid3) [reply keyboards/7splus/keymaps/salicylic/keymap.c:1]: Thank you. I forgot about it!

**Review by drashna — APPROVED** [review]

**Pull request merged**